geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/collections/IndexibleTreeSet.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.redis.internal.collections;
+
+import java.util.Iterator;
+import java.util.TreeSet;
+
+/**
+ * An OrderStatisticsSet implemented using TreeSet. {@link #indexOf(Object)} and other
+ * rank operations will be O(N) instead of O(log(N))
+ * Only for testing and performance comparisons.
+ */
+class IndexibleTreeSet<E> extends TreeSet<E> implements OrderStatisticsSet<E> {
```

Comment:
It doesn't need to be tested, I don't think, it just prevents possible serialization/backwards compatibility issues if the class is changed in the future (see https://issues.apache.org/jira/browse/GEODE-8996). If you don't explicitly specify an ID, one is auto-generated based on the contents of the class, so if the class changes, you can end up with two versions with different IDs which are incompatible.

Admittedly, this isn't a class we ever expect to be sending over the wire, particularly not between servers with different versions, but it doesn't cost much to include the ID just in case. There's actually an inspection in IntelliJ you can enable for this in Preferences -> Editor -> Inspections and then searching for "serialVersion" that will highlight when a class is missing one and will allow you to auto-generate one with a click.